include/swift/Basic/STLExtras.h

```diff
@@ -71,40 +71,6 @@ struct function_traits<R (T::*)(Args...) const> {
 
 /// @{
 
-/// An STL-style algorithm similar to std::for_each that applies a second
-/// functor between every pair of elements.
-///
-/// This provides the control flow logic to, for example, print a
-/// comma-separated list:
-/// \code
-///   interleave(names.begin(), names.end(),
-///              [&](StringRef name) { OS << name; },
-///              [&] { OS << ", "; });
-/// \endcode
-template <typename ForwardIterator, typename UnaryFunctor,
-          typename NullaryFunctor>
-inline void interleave(ForwardIterator begin, ForwardIterator end,
-                       UnaryFunctor each_fn,
-                       NullaryFunctor between_fn) {
-  if (begin == end)
-    return;
-  each_fn(*begin);
-  ++begin;
-  for (; begin != end; ++begin) {
-    between_fn();
-    each_fn(*begin);
-  }
-}
-
-template <typename Container, typename UnaryFunctor, typename NullaryFunctor>
-inline void interleave(const Container &c, UnaryFunctor each_fn,
-                       NullaryFunctor between_fn) {
-  interleave(c.begin(), c.end(), each_fn, between_fn);
-}
```

Comment:
Could we merge this change into `master` instead? Move this function into the LLVM namespace in this file, and then update all callers to use `llvm::` qualification. Then on the `master-next` branch, only delete the redundant implementation.

Reply:
See pull request https://github.com/apple/swift/pull/31091.

Reply:
I've removed the `llvm::interleave` changes from this pull request. Let's continue any discussion about this on https://github.com/apple/swift/pull/31091.